### PR TITLE
Switch to composition layer API that allows multiple layers.

### PR DESCRIPTION
**Pull request by dsnopek**
This adapts the passthrough extension wrapper for the GDExtension API changes in PR https://github.com/godotengine/godot/pull/89460

~~Leaving this marked as a draft until that PR is merged.~~

### CODE REVIEW

**Comment by dsnopek** [comment]
The upstream PR has been merged!

File: common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp
Comment (m4gr3d) [inline common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp:228]: nit: Does it make sense to use `in front of` / `behind` instead of `below` / `above` when describing the layer ordering?

Comment (dsnopek) [reply common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp:228]: Hm, yeah, I guess "behind" is a little clearer! Updated in latest push

File: common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp
Comment (m4gr3d) [inline common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp:228]: I am concerned starting with a high number would set a bad precedent to the rest of the community. 

I think instead we should decide on a range based on use-cases and constrain layer ordering within that range with clearly define constants to indicate the ordering of core capabilities. 

To avoid blocking this PR, I'd suggest starting with -100 instead (we have until v3.0 is publicly release to update that value if needed), and let's add the proposal as a topic of discussion.

**Review by m4gr3d — APPROVED** [review]

File: common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp
Comment (dsnopek) [reply common/src/main/cpp/extensions/openxr_fb_passthrough_extension_wrapper.cpp]: Makes sense! Updated :-)

(FYI, I took the `-9999` from [here](https://github.com/godotengine/godot/pull/76877/files#diff-40ce218497b07758c62e07b8cf35e4dfe5d5f3029e27ad3a1f9cc27d942bdd2fR169), which actually defaults all other layers to `1`, which is what I've continued using in my continued work on composition layers.)

**Pull request merged**
